source/LifecycleMethod.cpp

```diff
@@ -248,7 +248,55 @@ bool LifecycleMethod::validate(
   } else {
     // TODO:handle graph
     const auto& graph = std::get<LifeCycleMethodGraph>(body_);
-    static_cast<void>(graph); // hide unused variable warning.
+    if (graph.get_node("entry") == nullptr) {
+      ERROR(
+          1,
+          "Entry point entry is not a valid node in the lifecycle graph.");
+      return false;
+    }
+
+    for (const auto& [node_name, node] : graph.get_nodes()) {
+        for (const auto& method_call : node.method_calls()) {
+          method_call.validate(base_class, class_hierarchies);
+        }
+
+        for (const auto& successor_name : node.successors()) {
+          if (graph.get_node(successor_name) == nullptr) {
+            ERROR(
+                1,
+                "Node `{}` has a successor `{}` that is not a valid node in the lifecycle graph.",
+                node_name,
+                successor_name);
+            return false;
+          }
+        }
+      }
+
+    std::unordered_set<std::string> visited;
+    std::stack<std::string> stack;
+    stack.push("entry");
+
+    while (!stack.empty()) {
+      const std::string current_node = stack.top();
+      stack.pop();
+
+      if (visited.count(current_node) == 0) {
+        visited.insert(current_node);
+        const auto* node = graph.get_node(current_node);
+        for (const auto& successor : node->successors()) {
+          stack.push(successor);
+        }
+      }
+    }
+
+    // Check if all nodes were visited
+    if (visited.size() != graph.get_nodes().size()) {
+      ERROR(
+          1,
+          "Not all nodes are reachable from the entry point entry in the lifecycle graph.");
+      return false;
+    }
+
```

Comment:
could we move all this in a `validate` method fo `LifecycleMethodGraph`?

Reply:
you mean create `LifeCycleMethodGraph::validate` and in the current `LifecycleMethod::validate` call the one of the graph?